File: P1-E3/MotorcycleBuilder.java
package EJ3;

public class MotorcycleBuilder extends VehicleBuilder {
    public MotorcycleBuilder() {
        createNewVehicle();
    }

    public void buildEngine(String engine) {
        vehicle.engine = engine;
    }

    public void buildWheels() {
        vehicle.wheels = 2;
    }

    public void buildColor(boolean color, String color_vehicle) {
        vehicle.color = color;
        vehicle.color_vehicle = color_vehicle;
    }

    public void buildAudioSystem(boolean audioSystem) {
        vehicle.audioSystem = audioSystem;
    }

    public void buildTransmission(boolean transmission) {
        vehicle.transmission = transmission;
    }
}

File: P4/MotorcycleBuilder.dart
// ignore_for_file: file_names, non_constant_identifier_names, use_super_parameters

import 'VehicleBuilder.dart';

//Tengo que crear la clase MotorcycleBuilder abstracta para no implementar algunos metodos de la superclase
class MotorcycleBuilder extends VehicleBuilder{
  
//No estoy seguro de si se llama así con super. Faltaría el baseCost
  MotorcycleBuilder({required double baseCost}) : super(baseCost: baseCost){
    createNewVehicle();
    vehicle.tipo = "Moto";
  }

  @override
  void buildEngine(String engine){
    vehicle.engine = engine;
  }

  @override
  void buildWheels(){
    vehicle.wheels = 2;
  }

  @override
  void buildAudioSystem(bool audioSystem){
    vehicle.audioSystem = audioSystem;
  }

  @override
  void buildTransmission(bool transmission){
    vehicle.transmission = transmission;
  }
}